src/consumer/offsetManager/index.js

```diff
@@ -85,6 +92,32 @@ module.exports = class OffsetManager {
       .toString()
   }
 
+  /**
+   * @returns {Long}
+   */
+  countResolvedOffsets() {
+    const toPartitions = topic => keys(this.resolvedOffsets[topic])
+
+    const subtractOffsets = (resolvedOffset, committedOffset) => {
+      const resolvedOffsetLong = Long.fromValue(resolvedOffset)
+      return committedOffset === '-1'
```

Comment:
Is this right? I'm assuming this means that we have not committed any offsets, but `clearOffsets` will actually make  `this.committedOffsets[topic][partition]` undefined, right, so this would not be true.

Reply:
Before each `fetch` the consumer group will first resolve the offsets
https://github.com/tulios/kafkajs/blob/e65522ede62f3f8c0cf65ecabf715968ed389474/src/consumer/consumerGroup.js#L243

In `offsetManager#resolveOffsets` any invalid offset will be fetched from Kafka (`undefined`)
https://github.com/tulios/kafkajs/blob/e65522ede62f3f8c0cf65ecabf715968ed389474/src/consumer/offsetManager/index.js#L204

so `countResolvedOffsets` should be fine